torch/fx/experimental/symbolic_shapes.py

```diff
@@ -869,6 +877,8 @@ def _maybe_guard_eq(self, expr: "sympy.Eq") -> None:
                 except NotImplementedError:
                     pass
             return
+        except RecursionError:
+            raise RuntimeError(f"RecursionError in sympy.solve({lhs} - {rhs}, {free[0]})")
```

Comment:
```suggestion
        except RecursionError as e:
            raise RuntimeError(f"RecursionError in sympy.solve({lhs} - {rhs}, {free[0]})") from e
```

Reply:
This is actually intentional, usually the RecursionError is long and annoying so I felt like I don't actually want it haha

Reply:
Ah, I thought it might be useful for tracking down the recursion error.